test-data/unit/check-basic.test

```diff
@@ -75,6 +75,14 @@ if int():
     x = 1
 
 
+[case testIncompatibleAssignmentAmbiguousShortnames]
+
+class Any: pass
+
+x = Any()
+x = 1  # E: Incompatible types in assignment (expression has type "int", variable has type "__main__.Any")
```

Comment:
It would be good to test a few other type names as well, such as `list` / `List`. (Lower-case `list` might not work, since it's not in the `TYPES_FOR_UNIMPORTED_HINTS` set.)

Reply:
I've added a few more types from `TYPES_FOR_UNIMPORTED_HINTS`.